File: examples/javascript/transfer.js
#!/usr/bin/env node
/**
 * Example 1 — Native transfer (QCore.js)
 *
 * Send TQTOV from one Quantova account to another. Shows reading the head, the
 * sender account and nonce, the chain params for the fee, and the shape of an
 * ML-DSA-65 signed transfer submitted through the gateway.
 *
 * Run:
 *   export QUANTOVA_GATEWAY=http://127.0.0.1:8645   // or https://testnet.quantova.io
 *   node transfer.js
 *
 * In a full app with QCore.js installed:
 *
 *   const { QCore } = require('@qunatovainc/qcore');
 *   const q = new QCore(process.env.QUANTOVA_GATEWAY);
 *   const sender = q.wallet.create();                         // ML-DSA-65 account, Q1 address
 *   const signed = q.wallet.signTransfer(sender, {
 *     to: RECIPIENT, amount: 1_500_000n });                   // amount in Quon
 *   const hash = await q.submitTransaction(signed);
 *
 * This runnable version calls the gateway directly (see _common.js) so it works
 * without the @qunatovainc/qcore package installed.
 */
'use strict';
const { gw, toQtov, toQuon, banner } = require('./_common');

const SENDER = 'Q1GZD3AGFY5U426V9NX6UNE06ZC4YVKNK3GU9L3C';     // example Q1 address
const RECIPIENT = 'Q1GP3FR9CVK3XAJSQGC7DU47WCDUVTFDYMY0H82C';   // example Q1 address
const AMOUNT_QTOV = 1.5;

async function main() {
  banner('Quantova example: native TQTOV transfer (QCore.js)');

  const head = await gw('head');
  console.log(`Head height: ${head.height}`);

  const account = await gw('get_account', { address: SENDER });
  console.log(`Sender:  ${SENDER}`);
  console.log(`Balance: ${toQtov(account.balance || 0)} TQTOV   nonce: ${account.nonce || 0}`);

  // fee comes from the chain parameters
  const params = await gw('chain_params');
  console.log(`Fee:     base ${params.base_fee || params.min_fee || 'n/a'} Quon`);

  const amount = toQuon(AMOUNT_QTOV);
  console.log(`\nTransfer: ${AMOUNT_QTOV} TQTOV -> ${RECIPIENT}  (amount ${amount} Quon)`);
  console.log('In QCore.js the wallet builds and ML-DSA-65 signs this transfer, then submits it.');

  // In a full app, submit the signed transaction bytes here:
  //   const hash = (await gw('submit_transaction', { transaction: signedHex })).hash;
  console.log('\nSubmit with: q.submit_transaction({ transaction: <ML-DSA-65 signed hex> })');
  console.log('Then read it back with: q.get_transaction({ hash })');
}

main().catch((e) => { console.error('error:', e.message); process.exit(1); });

File: examples/python/transfer.py
#!/usr/bin/env python3
"""Example 1 — Native transfer (QCore.py)

Send TQTOV from one Quantova account to another. Shows connecting, reading a
balance and nonce, reading the fee from the chain parameters, deriving an ML-DSA-65
account, and signing the transfer before submitting it through the gateway.

Run:
    export QUANTOVA_GATEWAY=http://127.0.0.1:8645   # or https://testnet.quantova.io
    python transfer.py

This example registers a DEMO post quantum backend so it runs end to end offline.
In a real app you register Quantova's native Q-Crypto backend instead, the wallet
API is identical.
"""

from _common import banner, connect, to_quon, to_qtov, use_demo_backend

RECIPIENT = "Q1GP3FR9CVK3XAJSQGC7DU47WCDUVTFDYMY0H82C"  # example Q1 address, replace with a real one
AMOUNT_QTOV = 1.5


def main():
    banner("Quantova example: native TQTOV transfer (QCore.py)")
    use_demo_backend()  # demo only, real apps register the native Q-Crypto backend
    q = connect()

    head = q.head()
    print(f"Head height: {head['height']}")

    # derive an ML-DSA-65 sender account, its Q1 address begins with a capital Q
    sender = q.wallet.create()
    print(f"Sender (ML-DSA-65): {sender.address}")

    account = q.get_account(sender.address)
    print(f"Balance: {to_qtov(account.get('balance', 0))} TQTOV   nonce: {account.get('nonce', 0)}")

    # fee comes from the chain parameters
    params = q.chain_params()
    print(f"Base fee: {params.get('base_fee', params.get('min_fee', 'n/a'))} Quon")

    # build the transfer and sign it with ML-DSA-65
    amount = to_quon(AMOUNT_QTOV)
    signed = q.wallet.sign_transfer(sender, to=RECIPIENT, amount=amount)
    print(f"\nTransfer: {AMOUNT_QTOV} TQTOV -> {RECIPIENT}  (amount {amount} Quon)")
    print(f"ML-DSA-65 signature: {signed.signature_hex[:34]}...")

